### ISSUE TITLE
Suggestion for minimal package?

### ISSUE THREAD

**Issue by lirantal** [mallocator/versioned-api-router#1]
Ravi,

For someone who would just be interested in an API versioning capability there's a mouthful of functionality in the current package as it is and I was wondering if you'd be interested to break it up to another package?

Would be great to support multiple versions for the same route as well as defaulting to one of the versions (latest or oldest) if no version is specified in the request.

**Comment by mallocator** [comment]
Hi there,

Thanks for the suggestions, funny thing is the version router used to be all on its own. I had trouble getting it to work with the api-router and so I just pulled them both together, which brings me to a big caveat: This router isn't a fully compatible express router.
Unfortunately I couldn't find any documentation on express routers and so I had to work around what I could get from their code to get it working. The end product is a wrapper around 1+ Routers, which explains why not all methods are implemented.
If you're planning to use the router in production anyways, I recommend testing it well before hand. That's the stage I'm in, with the router itself not yet being in production.

If I ever figure out how to write my own router properly I will probably dissect this project and have the individual features separated, as it should be, but for now I can I only say:
